data/templates/template.typst

```diff
@@ -32,9 +33,17 @@
   set heading(numbering: sectionnumbering)
 
   if title != none {
-    align(center)[#block(inset: 2em)[
-      #text(weight: "bold", size: 1.5em)[#title]
-    ]]
+    if subtitle != none {
+       align(center)[#block(inset: 2em)[
+        #text(weight: "bold", size: 1.5em)[#title]
+        #parbreak()
+        #text(weight: "bold", size: 1.25em)[#subtitle]
+      ]]
+    } else {
+      align(center)[#block(inset: 2em)[
+        #text(weight: "bold", size: 1.5em)[#title]
+      ]]
+    }
```

Comment:
Seems like we could simplify this by moving the conditional inside:
```
align(center)(block(inset: 2em)[
  #text(weight: "bold", size: 1.5em)[#title] 
  #(if subtitle != none {
    #parbreak() etc.
   })
])
```
(untested)

Reply:
The right syntax would be:

```
  if title != none {
    align(center)[#block(inset: 2em)[
      #text(weight: "bold", size: 1.5em)[#title]
      #(if subtitle != none {
        parbreak()
        text(weight: "bold", size: 1.25em)[#subtitle]
      })
    ]]
  }
```

Although this makes the tests fail with a very cryptic "error":

```txt
pandoc-lua-engine> Test suite test-pandoc-lua-engine passed
pandoc           > pandoc tests
pandoc           >   Old
pandoc           >     typst
pandoc           >       writer
pandoc           >         basic: FAIL (0.09s)
pandoc           >           
pandoc           >           ------------------------------------------------------------------------
pandoc           >           --- writer.typst
pandoc           >           +++ /Users/mcanouil/Projects/misc/pandoc/.stack-work/dist/aarch64-osx/ghc-9.6.4/build/test-pandoc/test-pandoc --emulate --quiet testsuite.native -r native -w typst --columns=80 --variable pandoc-version= -s
pandoc           >           +  66   }
pandoc           >           ------------------------------------------------------------------------
pandoc           >           Use -p '/tests.Old.typst.writer.basic/&&$0=="pandoc tests.Old.typst.writer.basic"' to rerun this test only.
pandoc           >         basic: FAIL (0.07s)
pandoc           >           
pandoc           >           ------------------------------------------------------------------------
pandoc           >           --- writer.typst
pandoc           >           +++ /Users/mcanouil/Projects/misc/pandoc/.stack-work/dist/aarch64-osx/ghc-9.6.4/build/test-pandoc/test-pandoc --emulate --quiet testsuite.native -r native -w typst --columns=80 --variable pandoc-version= -s
pandoc           >           +  66   }
pandoc           >           ------------------------------------------------------------------------
pandoc           >           Use -p '/tests.Old.typst.writer.basic/&&$0=="pandoc tests.Old.typst.writer.basic"' to rerun this test only.
pandoc           >  
pandoc           > 2 out of 2 tests failed (0.16s)
pandoc           > Test suite test-pandoc failed
Completed 2 action(s).
```

Reply:
Hm.  Could this be a line-ending issue?
Worth running `/Users/mcanouil/Projects/misc/pandoc/.stack-work/dist/aarch64-osx/ghc-9.6.4/build/test-pandoc/test-pandoc --emulate --quiet testsuite.native -r native -w typst --columns=80 --variable pandoc-version= -s` and inspecting the output carefully (including line endings), comparing to the expected test output.

Reply:
Resolved in https://github.com/jgm/pandoc/pull/9747/commits/ab768863c7150494036eac679ddb7517237edc1c